src/lxc/tools/lxc_unshare.c

```diff
@@ -241,12 +254,43 @@ int main(int argc, char *argv[])
 		exit(EXIT_FAILURE);
 	}
 
+	if (start_arg.setuid) {
+		wait_fd = eventfd(0, EFD_CLOEXEC);
+		if (wait_fd == -1) {
+			perror("eventfd");
+			exit(EXIT_FAILURE);
+		}
+	}
+
 	pid = lxc_clone(do_start, &start_arg, flags);
 	if (pid < 0) {
 		fprintf(stderr, "failed to clone\n");
 		exit(EXIT_FAILURE);
 	}
 
+	if (start_arg.setuid) {
+		/* enough space to accommodate uids */
+		char *umap = (char *)alloca(100);
+
+		/* create new uid mapping using current UID and the one
+		 * specified as parameter
+		 */
+		if (snprintf(umap, 100, "%d %d 1" , *(start_arg.uid), getuid()) < 0) {
+			fprintf(stderr, "snprintf failed");
+			exit(EXIT_FAILURE);
+		}
+
+		if (write_id_mapping(ID_TYPE_UID, pid, umap, strlen(umap))) {
```

Comment:
Iirc, writing to `/proc/<pid>/uid_map` requires the string to include a final `\n`.